src/Razor/src/Microsoft.VisualStudioCode.RazorExtension/Services/RazorDynamicFileUpdate.cs

```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT license. See License.txt in the project root for license information.
+
+using System.Text.Json.Serialization;
+
+namespace Microsoft.VisualStudioCode.RazorExtension.Services;
+
+internal sealed class RazorDynamicFileUpdate
+{
+    [JsonPropertyName("edits")]
+    public required ServerTextChange[] Edits { get; set; }
```

Comment:
We have `RazorTextChange` which looks the same as `ServerTextChange`. Also, why not put this array directly in `RazorProvideDynamicFileResponse`? Looks like there is even a debug assert to ensure there is only ever one `RazorDynamicFileUpdate` anyway